service/controller/app/v1/resource/kubeconfigfinalizer/create.go

```diff
@@ -0,0 +1,67 @@
+package kubeconfigfinalizer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if key.InCluster(cr) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("In-cluster used"))
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no need to put the finalizer"))
```

Comment:
I do not understand what this is about. What is "in cluster"? Further, the "canceling resource" log is missing.

Reply:
`In-cluster` is for when you would like to install `chart` in CP, not in TC. 
https://github.com/giantswarm/giantswarm/blob/master/roadmap/MANAGED-SERVICES-CATALOG.MD#kubeconfig

Reply:
So when it is "in cluster" there is no secret to be maintained during app deletion?

Reply:
Yes, we don't have any kubeconfig secrets in `in-cluster` case.

Reply:
Ok then I would like to have better logging and maybe even a comment in the code to explain all of this in a couple of sentences.